internal/witness/cmd/feeder/main.go

```diff
@@ -53,14 +70,18 @@ type ctLog struct {
 
 // populateLogs populates a list of ctLogs based on the log list.
 func populateLogs(logListURL string) ([]ctLog, error) {
-	resp, err := http.Get(logListURL)
+	u, err := url.Parse(logListURL)
 	if err != nil {
-		return nil, fmt.Errorf("failed to retrieve log list: %v", err)
+		return nil, fmt.Errorf("failed to parse URL: %v", err)
 	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
+	s := u.Scheme
+	queryFn, ok := getByScheme[s]
```

Comment:
Do you think it might be worth extracting this stuff out into a "readURL" func? it might a bit more explicit and contained?

Could _potentially_ pop that func into an `internal` dir for the witness and share across `main` files which need this functionality?

Reply:
Sure, done on the `readURL` function!  I think moving the map and functions into a whole new dir is a bit overkill for now?